libraries/scopes/test/simplify_test.js

```diff
@@ -0,0 +1,13 @@
+const assert = require('assert');
+const utils = require('../src');
+const testing = require('taskcluster-lib-testing');
+
+suite(testing.suiteName(), function() {
+  test('Simplify Scope Expression test ', function() {
+    assert(utils.simplifyScopeExpression('First test'));
```

Comment:
I think you'll want to use `assert.deepEqual`.  Here's an example test:

```js
test('single scope simplifies to itself', function() {
  assert.equal(simplifyScopeExpression("scope:1"), "scope:1");
});
```
then update the simplifyScopeExpression function to pass that test.  Then add another test.  It's good to start with the "base cases" for any recursive data structure.  From [here](https://github.com/taskcluster/taskcluster/tree/master/libraries/scopes#new-style) we can see that the single scope is one of three options, with the other two being AnyOf and AllOf.  And for the moment we can stick to cases where no simplification is possible.

```js
test('AnyOf simplifies to itself', function() {
  assert.equal(
    simplifyScopeExpression({AnyOf: ["scope:1", "scope:2"]}),
    {AnyOf: ["scope:1", "scope:2"]});
});

test('AllOf simplifies to itself', function() {
  assert.equal(
    simplifyScopeExpression({AllOf: ["scope:1", "scope:2"]}),
    {AllOf: ["scope:1", "scope:2"]});
});
```

Once you've got that working, here are a few more advanced tests to work on:

```js
test('AnyOf list of scopes duplicates are removed', function() {
  assert.equal(
    simplifyScopeExpression({AnyOf: ["scope:1", "scope:2", "scope:1"]}),
    {AnyOf: ["scope:1", "scope:2"]});
});

test('AllOf list of scopes duplicates are removed', function() {
  assert.equal(
    simplifyScopeExpression({AllOf: ["scope:1", "scope:2", "scope:1"]}),
    {AllOf: ["scope:1", "scope:2"]});
});

test('AllOf list of scopes is normalized', function() {
  // note that AnyOf lists of scopes should *not* be normalized this way!
  assert.equal(
    simplifyScopeExpression({AllOf: ["scope:1", "scope:2", "scope:*"]}),
    {AllOf: ["scope:*"]});
});
```